set-matter.sh

```diff
@@ -71,7 +76,17 @@ if [ "$UID" -eq "$ROOT_UID" ]; then
       echo -e "Setting theme to ${PALETTE}"
       sed -i -E "s/(selected_item_color = )\"#[0-9a-fA-F]+\"/\1\"${!PALETTE}\"/" "${THEME_DIR}/${THEME_NAME}/theme.txt"
   fi
-  
+
+  # Setting available fonts
+  # List of supported fonts
+  supported_fonts=(poiret-one karla pattaya)
+
+  # Set the chosen font
+  if [[ ! " ${FONT} " == "  " ]] && [[ " ${supported_fonts[@] " =~ " ${FONT} " ]]; then
+      echo -e "Setting font to ${FONT}"
+      grub-mkfont "./Matter/fonts/${FONT}/${FONT}-regular.ttf"
```

Comment:
we need to supply an output path like `grub-mkfont -s 20 -o /Matter/${FONT}.pf2 "./Matter/fonts/${FONT}/${FONT}-regular.ttf"`

But even if we do so, and change the contents of `theme.txt` by `sed` (as is done for palette colors), still the generated `pf2` font file will remain in the current directory (not copied to the `/boot/grub/themes/Matter` dir) because the `Matter` dir is being copied to the `/boot/grub/themes/Matter` dir even before the palette colors are changed. It is being changed directly into the `theme.txt` which is present in the `/boot/grub/themes/Matter` dir by `sed`.

What I have done so far (because I don't know how to do it) is to manually copy the generated `pf2` file after the Matter dir has been copied to `/boot/grub/themes/` dir and replaced the font in `theme.txt` manually. This works on my system but it should really be incorporated into this script.
Therefore, the things to be done may include -

-  adding support for both `-p` and `-f` options
- for -f option, generate and copy the `pf2` file to the `/boot/grub/themes/Matter` dir
- change font in `theme.txt`

Reply:
This should be solved by moving the `# Copy theme` just above `# Set theme` and options like `-p` and -`f` should modify the local `Matter` folder before moving it to `/boot/grub/themes`